src/main.rs

```diff
@@ -0,0 +1,230 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use futures::StreamExt;
+use std::io;
+use typedb_client::{
+    concept::{Attribute, Concept, Value},
+    Connection, DatabaseManager, Session,
+    SessionType::Data,
+    TransactionType::Read,
+};
+mod common;
+
+#[derive(Debug)]
+enum HandleError {
+    Io(io::Error),
+    TypeDB(typedb_client::error::Error),
+}
+
+const MENU_DATABASE: &str = "menuDB1";
+fn read_input() -> Result<String, HandleError> {
+    let mut input_string = String::new();
+    io::stdin()
+        .read_line(&mut input_string)
+        .map_err(HandleError::Io)?;
+    Ok(input_string.trim().to_string())
+}
+
+async fn get_sellers(connection: Connection) -> Result<(), HandleError> {
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(
+        databases
+            .get(MENU_DATABASE)
+            .await
+            .map_err(HandleError::TypeDB)?,
+        Data,
+    )
+    .await
+    .map_err(HandleError::TypeDB)?;
+    let transaction = session
+        .transaction(Read)
+        .await
+        .map_err(HandleError::TypeDB)?;
+
+    println!("Enter the name of raw_food x : ");
+    let x: String = read_input()?;
+    let q = format!(
+        "match $rf isa raw_food, has name \"{x}\";
+        $is_i (raw_food: $rf,dish: $d) isa is_ingredient;
+        $sl (seller: $m,product: $d) isa sells;
+        $m has name $n;
+        get $n;"
+    );
+    let mut answer_stream = transaction
+        .query()
+        .match_(q.as_str())
+        .map_err(HandleError::TypeDB)?;
+    while let Some(result) = answer_stream.next().await {
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute {
+                        value: Value::String(value),
+                        ..
+                    }) = concept
+                    {
+                        println!("{}", value);
+                    }
+                }
+            }
+            Err(err) => {
+                return Err(HandleError::TypeDB(err));
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn get_strange_menu(connection: Connection) -> Result<(), HandleError> {
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(
+        databases
+            .get(MENU_DATABASE)
+            .await
+            .map_err(HandleError::TypeDB)?,
+        Data,
+    )
+    .await
+    .unwrap();
+    let transaction = session
+        .transaction(Read)
+        .await
+        .map_err(HandleError::TypeDB)?;
+
+    println!("::Q2::");
+    let q = "
+    match $m isa menu, has is_vegetarian false,has name $n;
+    $d isa dish, has is_vegetarian false;
```

Comment:
This query doesn't quite do what you say it does in the query runner prompt. You'll need to change the dish to be vegetarian rather than not vegetarian. 
```suggestion
    $d isa dish, has is_vegetarian true;
```